data-prepper-plugins/aggregate-processor/src/test/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateProcessorTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.configuration.PluginModel;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.model.plugin.PluginFactory;
+import com.amazon.dataprepper.model.record.Record;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class AggregateProcessorTest {
+
+    @Mock
+    private PluginFactory pluginFactory;
+
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private AggregateIdentificationKeysHasher aggregateIdentificationKeysHasher;
+
+    @Mock
+    private AggregateProcessorConfig aggregateProcessorConfig;
+
+    @Mock
+    private AggregateAction aggregateAction;
+
+    @Mock
+    private PluginModel actionConfiguration;
+
+    @Mock
+    private GroupStateManager groupStateManager;
+
+    @Mock
+    private AggregateActionResponse aggregateActionResponse;
+
+    private AggregateProcessor aggregateProcessor;
```

Comment:
You don't need this field variable. Using local variables in tests is more maintainable.

- [ ] Remove unnecessary testing field for object under test.